src/test/java/com/training/api/tests/cat/CatApiTests.java

```diff
@@ -1,79 +1,91 @@
 import org.junit.jupiter.params.provider.CsvFileSource;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
-import org.json.simple.JSONObject;
-import org.junit.jupiter.api.Assertions;
-import java.util.List;
-import java.util.Map;
+
+import java.util.*;
 
 public class CatApiTests {
-    private Logger logger = LoggerFactory.getLogger(this.getClass());
     protected Map<String, String> map = new JsonReaderUtility().getMap();
-    RequestSpecification httpRequest;
+    Map<String, Object> headerMap;
 
     @BeforeEach
     public void setup() {
         RestAssured.baseURI = map.get("baseUri");
-        httpRequest = RestAssured.given();
-        httpRequest.header(map.get("catAuthenticationHeaderName"), map.get("catAuthenticationHeaderValue"));
+        headerMap = new HashMap();
+        headerMap.put(map.get("catAuthenticationHeaderName"), map.get("catAuthenticationHeaderValue"));
     }
 
     @ParameterizedTest
     @CsvFileSource(resources = "/getIdResponse.csv")
-    public void testGetIdResponse(String name, String breedId) {
+    public void testGetBreedId(String name, String breedId) {
+        String id = "";
+        io.restassured.response.Response response = new RestEngine().getResponse("/v1/breeds", headerMap);
+        Assertions.assertEquals(200, response.getStatusCode(), "Request Unsuccessful");
 
-        Response response = httpRequest.get("/v1/breeds");
 
-        Assertions.assertEquals(response.getStatusCode(), 200);
+        for (com.training.pojos.cat.breeds.Response responseElement :
+                Arrays.asList(response.as(com.training.pojos.cat.breeds.Response[].class))) {
+            if (responseElement.getName().equalsIgnoreCase(name)) {
+                id = responseElement.getId();
+                break;
+            }
+        }
 
-        List<String> breedName = response.jsonPath().getList("name");
-        int index = breedName.indexOf(name);
-        String id = response.jsonPath().getString(String.format("id[%s]",index));
-        logger.info(id);
-        Assertions.assertEquals(breedId, id,"Id incorrect");
+        Assertions.assertEquals(id, breedId, "Id incorrect");
     }
 
     @ParameterizedTest
     @CsvFileSource(resources = "/getWikipediaUrl.csv")
     public void testGetWikipediaUrl(String id, String url) {
+        io.restassured.response.Response response = new RestEngine()
+                .getResponse(String.format("/v1/images/search?breed_ids=%s", id), headerMap);
+
+        Assertions.assertEquals(200, response.getStatusCode(), "Request Unsuccessful");
 
-        String path = String.format("/v1/images/search?breed_ids=%s", id);
-        Response response = httpRequest.when().get(path);
-        Assertions.assertEquals( 200,response.getStatusCode(),"Request Unsuccessful");
+        List<com.training.pojos.cat.search.Response> responseList =
+                Arrays.asList(response.as(com.training.pojos.cat.search.Response[].class));
```

Comment:
use this directly in assertion, mention this in previous comments ,hehe

Reply:
It is in assertion. i hadn't removed this line